### PR TITLE
JAPI-144 wsfileio array copy enhancement

### PR DESCRIPTION
**Pull request by rpastrana**
- Utilize java.arrays function to copy byte array
- Add junit

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [ ] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java [hunk 1]
```diff
@@ -204,18 +205,16 @@ public boolean writeHPCCFileData(byte []  data, String fileName, String targetLa
...
         while (bytesleft > 0)
         {
             payloadsize = bytesleft >= limit ? limit : bytesleft;
-            log.trace("Writing offset: "+dataindex+"\t size: "+((int)dataindex + (int)payloadsize));
+            log.trace("Writing offset: "+dataindex+"\t size: "+(dataindex + payloadsize));
 
-            subdata = new byte [(int)payloadsize];
-            for (int i = 0; i < payloadsize ; i++, dataindex++)
-                subdata[i] = data[dataindex];
+            subdata = Arrays.copyOfRange(data, 0, payloadsize-1);
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java]: It looks like we aren't changing the copy offset. I think this should be something like:
`
int end = dataindex + payloadsize-1;
subdata = Arrays.copyOfRange(data, dataindex, end);
dataindex += payloadsize;
`

**Comment by rpastrana** [comment]
@jpmcmu please review

**Comment by rpastrana** [comment]
@jpmcmu bump

**Comment by jpmcmu** [comment]
@rpastrana looks good

**Pull request merged**